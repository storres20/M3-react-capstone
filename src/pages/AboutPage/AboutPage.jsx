/* eslint-disable max-len */
import styles from './AboutPage.module.css';
import ScrollButton from '../../components/ScrollButton/ScrollButton';

export default function AboutPage() {
  return (
    <>
      <div className={styles.main}>
        <div className={`${styles.header} ${styles.gradient}`}>
          <div className={styles.inner_content}>
            <div className={styles.content}>
              <h2>Hi there,</h2>
              <img src="https://www.themoviedb.org/assets/2/v4/marketing/deadpool-06f2a06d7a418ec887300397b6861383bf1e3b72f604ddd5f75bce170e81dce9.png" width="451" height="149" alt="pictureT" />

              <h3>Let&lsquo;s talk about TMDB</h3>
              <p>
                The Movie Database (TMDB) is a
                <em>community built</em>
                {' '}
                movie and TV database. Every piece of data has been added by our amazing community dating back to 2008. TMDB&lsquo;s strong
                <em>international focus</em>
                {' '}
                and breadth of data is largely unmatched and something we&lsquo;re incredibly proud of. Put simply, we live and breathe community and that&lsquo;s precisely what makes us different.
              </p>

              <h4>The TMDB advantage</h4>
              <div className={styles.wrapper}>
                <div>
                  <div>1</div>
                  <p>Every year since 2008, the number of contributions to our database has increased. With over 750,000 developers and companies using our platform, TMDB has become a premiere source for metadata.</p>
                </div>

                <div>
                  <div>2</div>
                  <p>Along with extensive metadata for movies, TV shows and people, we also offer one of the best selections of high resolution posters and fanart. On average, over 1,000 images are added every single day.</p>
                </div>

                <div>
                  <div>3</div>
                  <p>We&lsquo;re international. While we officially support 39 languages we also have extensive regional data. Every single day TMDB is used in over 180 countries.</p>
                </div>

                <div>
                  <div>4</div>
                  <p>Our community is second to none. Between our staff and community moderators, we&lsquo;re always here to help. We&lsquo;re passionate about making sure your experience on TMDB is nothing short of amazing.</p>
                </div>

                <div>
                  <div>5</div>
                  <p>Trusted platform. Every single day our service is used by millions of people while we process over 3 billion requests. We&lsquo;ve proven for years that this is a service that can be trusted and relied on.</p>
                </div>
              </div>

              <a href="https://www.themoviedb.org/" target="_blank" className={`${styles.contact} ${styles.button} ${styles.white}`} rel="noreferrer">Contact TMDB</a>
            </div>
          </div>
        </div>
      </div>
      <ScrollButton />
    </>
  );
}
