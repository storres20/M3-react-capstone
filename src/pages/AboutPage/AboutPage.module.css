.main {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  align-content: flex-start;
  justify-content: center;
  background-color: #fff;
}

.header {
  background-position: 0 0;
  background-color: #0a1526;
}

.gradient {
  width: 100%;
  background-image: radial-gradient(at 25% top, #32112c 0%, #0a1526 40%);
  z-index: 0;
}

.inner_content {
  padding-top: 70px;
  box-sizing: border-box;
  width: 100%;
  background-image: url("https://www.themoviedb.org/assets/2/v4/marketing/red_pipes-594140f35efb300741add9827cc4f41053b0a4bb7ba249a40375377cc9d22d47.svg");
  background-position: center -250px;
  background-repeat: no-repeat;
  z-index: 2;
}

.content {
  color: #fff;
  padding: 20px 0 70px 0;
  width: 800px;
  margin: 0 auto;
  text-align: center;
}

.content > h2 {
  height: 233px;
  line-height: 173px;
  font-family: 'Lato', sans-serif;
  font-style: italic;
  font-size: 8em;
  font-weight: 400;
}

.content > img {
  margin-top: -140px;
  width: 100%;
  height: auto;
}

.content > h3 {
  font-size: 4em;
  margin-top: 40px;
  margin-bottom: 20px;
  font-weight: 700;
}

.content > p {
  margin: 0 auto;
  text-align: center;
  font-size: 1.2em;
}

.content > h4 {
  font-size: 2em;
  margin-top: 70px;
  margin-bottom: 40px;
  font-weight: 700;
}

.wrapper {
  margin: 0 auto;
  width: 100%;
}

.wrapper > div {
  width: 100%;
  display: flex;
  align-items: flex-start;
  margin-bottom: 20px;
}

.wrapper > div > div {
  color: #d40242;
  font-weight: 700;
  font-size: 4em;
  line-height: 0.8em;
  opacity: 0.7;
  width: 8vw;
}

.wrapper > div > p {
  text-align: left;
  margin: 0;
  width: 92vw;
}

.white {
  background-color: #fff;
  color: #d40242 !important;
}

.button {
  margin-top: 40px;
  border-radius: 5px;
  display: inline-block;
  padding: 10px 20px;
  text-transform: uppercase;
  font-weight: 700;
  font-size: 1.2em;
  transition: 0.2s all;
}

.button:hover {
  background-color: #d40242;
  color: #fff !important;
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (max-device-width: 991px) {
  /* Styles */
  .inner_content {
    /* width: 100hv; */
    background-position: right -350px;
  }

  .content {
    width: 100%;
  }

  .content > h2 {
    height: 143px;
    line-height: 33px;
    font-size: 4em;
    font-weight: 400;
  }

  .content > img {
    width: 90%;
    height: auto;
  }

  .content > h3 {
    font-size: 2em;
    margin-top: 40px;
    margin-bottom: 20px;
    font-weight: 700;
  }

  .content > p {
    margin: 0 20px;
  }

  .wrapper > div > div {
    width: 15%;
  }

  .wrapper > div > p {
    width: 80%;
    text-align: justify;
  }
}
