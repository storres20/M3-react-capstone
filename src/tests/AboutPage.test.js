import React from 'react';
import renderer from 'react-test-renderer';
import AboutPage from '../pages/AboutPage/AboutPage';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AboutPage />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
