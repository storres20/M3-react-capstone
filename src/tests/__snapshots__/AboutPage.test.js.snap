// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="main"
  >
    <div
      className="header gradient"
    >
      <div
        className="inner_content"
      >
        <div
          className="content"
        >
          <h2
            data-testid="title"
          >
            Hi there,
          </h2>
          <img
            alt="pictureT"
            height="149"
            src="https://www.themoviedb.org/assets/2/v4/marketing/deadpool-06f2a06d7a418ec887300397b6861383bf1e3b72f604ddd5f75bce170e81dce9.png"
            width="451"
          />
          <h3>
            Let‘s talk about TMDB
          </h3>
          <p
            data-testid="paragraph"
          >
            The Movie Database (TMDB) is a
            <em>
              community built
            </em>
             
            movie and TV database. Every piece of data has been added by our amazing community dating back to 2008. TMDB‘s strong
            <em>
              international focus
            </em>
             
            and breadth of data is largely unmatched and something we‘re incredibly proud of. Put simply, we live and breathe community and that‘s precisely what makes us different.
          </p>
          <h4>
            The TMDB advantage
          </h4>
          <div
            className="wrapper"
            data-testid="wrapper"
          >
            <div>
              <div>
                1
              </div>
              <p>
                Every year since 2008, the number of contributions to our database has increased. With over 750,000 developers and companies using our platform, TMDB has become a premiere source for metadata.
              </p>
            </div>
            <div>
              <div>
                2
              </div>
              <p>
                Along with extensive metadata for movies, TV shows and people, we also offer one of the best selections of high resolution posters and fanart. On average, over 1,000 images are added every single day.
              </p>
            </div>
            <div>
              <div>
                3
              </div>
              <p>
                We‘re international. While we officially support 39 languages we also have extensive regional data. Every single day TMDB is used in over 180 countries.
              </p>
            </div>
            <div>
              <div>
                4
              </div>
              <p>
                Our community is second to none. Between our staff and community moderators, we‘re always here to help. We‘re passionate about making sure your experience on TMDB is nothing short of amazing.
              </p>
            </div>
            <div>
              <div>
                5
              </div>
              <p>
                Trusted platform. Every single day our service is used by millions of people while we process over 3 billion requests. We‘ve proven for years that this is a service that can be trusted and relied on.
              </p>
            </div>
          </div>
          <a
            className="contact button white"
            href="https://www.themoviedb.org/"
            rel="noreferrer"
            target="_blank"
          >
            Contact TMDB
          </a>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="button"
    data-testid="ScrollButton"
  />,
]
`;
